src/app/subnets/hooks.ts

```diff
@@ -0,0 +1,17 @@
+import { useSelector } from "react-redux";
+
+import type { RootState } from "app/store/root/types";
+import subnetSelectors from "app/store/subnet/selectors";
+import vlanSelectors from "app/store/vlan/selectors";
+import type { VLAN, VLANMeta } from "app/store/vlan/types";
+
+export const useGetVlanSubnets = (id: VLAN[VLANMeta.PK] | null) => {
```

Comment:
New `subnet_ids` could simplify the implementation and potentially provide better performance in places such as below, where we use components that only take subnet id (e.g. `<SubnetLink />` ).

https://github.com/ndv99/maas-ui/blob/0c96619734b20712e1fb7d15ebf58fc2169a5e8c/src/app/subnets/views/VLANDetails/VLANSubnets/VLANSubnets.tsx#L59-L74

This would require deeper refactoring, which I'd say is out of scope of this.

There's nothing wrong with this other than it could only bring in confusion considering we already have equivalent utilities. For that reason I'd say let's remove this utility and focus on getting this merged.

Reply:
I've reverted the commit that implements this utility, looks like CI has passed.

Should we make a separate maintenance task for some deeper investigation/refactoring?

Reply:
Considering the amount of work it will require and how little value for the user it'll bring I'd say it's not a priority at the moment. I added a link to this conversation to https://bugs.launchpad.net/maas/+bug/1996500

Fixing this would involve reworking subnets tables and this information may be valuable.

Reply:
Sounds good to me. Let's leave it for now